habitat_baselines/evaluate_ppo.py

```diff
@@ -149,13 +234,120 @@ def main():
         episode_counts += 1 - not_done_masks
         current_episode_reward *= not_done_masks
 
+        # added for video support
+        next_episodes = envs.current_episodes()
+        envs_to_pause = []
+        n_envs = envs.num_envs
+        for i in range(n_envs):
+            if next_episodes[i].episode_id in stats_episodes:
+                envs_to_pause.append(i)
+
+            if not_done_masks[i].item() == 0:
+                # new episode ended, record stats
+                stats_episodes.add(current_episodes[i].episode_id)
+                if args.video == 1 and len(rgb_frames[i]) > 0:
+
+                    video_name = "{}{}_{}{}_{}{:.2f}".format(
+                        "episode",
+                        current_episodes[i].episode_id,
+                        "ckpt",
+                        checkpoint_idx,
+                        "spl",
+                        infos[i]["spl"],
+                    )
+
+                    images_to_video(
+                        rgb_frames[i], args.out_dir_video, video_name
+                    )
+
+                    images_to_tb_video(
+                        "episode{}".format(current_episodes[i].episode_id),
+                        checkpoint_idx,
+                        rgb_frames[i],
+                        writer,
+                        fps=10,
+                    )
+
+                    rgb_frames[i] = []
+
+            elif args.video == 1:
+                observation_size = observations[i]["rgb"].shape[0]
```

Comment:
This all should again go into separate file, so it is easy to use in other training and evaluation scripts.